### PR TITLE
Move Encoding serialization to the compiler layer

### PR DESCRIPTION
**Pull request by tmat**
Follow up on https://github.com/dotnet/roslyn/pull/45374

### CODE REVIEW

**Comment by tmat** [comment]
@jaredpar @CyrusNajmabadi @dotnet/roslyn-compiler  PTAL

**Comment by jaredpar** [comment]
@dotnet/roslyn-compiler PTAL

File: src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs [hunk 1]
```diff
@@ -1131,6 +1131,64 @@ private void TestRoundTripArray<T>(T[] values)
             TestRoundTripValue(values);
         }
 
+        [Theory]
+        [CombinatorialData]
+        public void Encoding_UTF8(bool byteOrderMark)
+        {
+            TextRoundripEncoding(new UTF8Encoding(byteOrderMark));
```
Comment (RikkiGibson) [inline src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs:1138]: nit: Te**s**tRound**t**ripEncoding?

Comment (tmat) [reply src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs:1138]: Yes! :)

File: src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs [hunk 2]
```diff
@@ -1135,33 +1135,33 @@ private void TestRoundTripArray<T>(T[] values)
         [CombinatorialData]
         public void Encoding_UTF8(bool byteOrderMark)
         {
-            TextRoundripEncoding(new UTF8Encoding(byteOrderMark));
+            TestRoundripEncoding(new UTF8Encoding(byteOrderMark));
```
Comment (RikkiGibson) [inline src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs:1138]:
```suggestion
            TestRoundtripEncoding(new UTF8Encoding(byteOrderMark));
```

File: src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs [hunk 3]
```diff
@@ -1135,33 +1135,33 @@ private void TestRoundTripArray<T>(T[] values)
...
         }
 
         [Theory]
         [CombinatorialData]
         public void Encoding_UTF32(bool bigEndian, bool byteOrderMark)
         {
-            TextRoundripEncoding(new UTF32Encoding(bigEndian, byteOrderMark));
+            TestRoundripEncoding(new UTF32Encoding(bigEndian, byteOrderMark));
```
Comment (RikkiGibson) [inline src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs:1145]:
```suggestion
            TestRoundtripEncoding(new UTF32Encoding(bigEndian, byteOrderMark));
```

File: src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs [hunk 4]
```diff
@@ -1135,33 +1135,33 @@ private void TestRoundTripArray<T>(T[] values)
...
         }
 
         [Theory]
         [CombinatorialData]
         public void Encoding_Unicode(bool bigEndian, bool byteOrderMark)
         {
-            TextRoundripEncoding(new UnicodeEncoding(bigEndian, byteOrderMark));
+            TestRoundripEncoding(new UnicodeEncoding(bigEndian, byteOrderMark));
```
Comment (RikkiGibson) [inline src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs:1152]:
```suggestion
            TestRoundtripEncoding(new UnicodeEncoding(bigEndian, byteOrderMark));
```

File: src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs [hunk 5]
```diff
@@ -1135,33 +1135,33 @@ private void TestRoundTripArray<T>(T[] values)
...
         }
 
         [Fact]
         public void Encoding_AllAvailable()
         {
             foreach (var info in Encoding.GetEncodings())
             {
-                TextRoundripEncoding(Encoding.GetEncoding(info.Name));
+                TestRoundripEncoding(Encoding.GetEncoding(info.Name));
```
Comment (RikkiGibson) [inline src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs:1160]:
```suggestion
                TestRoundtripEncoding(Encoding.GetEncoding(info.Name));
```

File: src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs [hunk 6]
```diff
@@ -1135,33 +1135,33 @@ private void TestRoundTripArray<T>(T[] values)
...
             }
         }
 
-        private static void TextRoundripEncoding(Encoding encoding)
+        private static void TestRoundripEncoding(Encoding encoding)
```
Comment (RikkiGibson) [inline src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs:1164]:
```suggestion
        private static void TestRoundtripEncoding(Encoding encoding)
```

**Pull request merged**